samples/matter/common/src/diagnostic/diagnostic_logs_provider.h

```diff
@@ -77,30 +79,11 @@ class DiagnosticLogProvider : public chip::app::Clusters::DiagnosticLogs::Diagno
 	DiagnosticLogProvider(DiagnosticLogProvider &&) = delete;
 	DiagnosticLogProvider &operator=(DiagnosticLogProvider &&) = delete;
 
-	/* Crash logs */
-#ifdef CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_CRASH_LOGS
-	CHIP_ERROR LoadCrashData(CrashData *crashData);
-	CHIP_ERROR GetCrashLogs(chip::MutableByteSpan &outBuffer, bool &outIsEndOfLog);
-	CHIP_ERROR FinishCrashLogs();
-	size_t GetCrashLogsSize();
-
-#ifdef CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_SAVE_CRASH_TO_SETTINGS
-	DiagnosticLogProvider()
-		: mDiagnosticLogsStorageNode("dl", strlen("dl")),
-		  mCrashLogsStorageNode("cl", strlen("cl"), &mDiagnosticLogsStorageNode)
-	{
-	}
-	CHIP_ERROR MoveCrashLogsToNVS();
-
-	Nrf::PersistentStorageNode mDiagnosticLogsStorageNode;
-	Nrf::PersistentStorageNode mCrashLogsStorageNode;
-#endif
-	Nrf::CrashData *mCrashData = nullptr;
-#endif
-
-#ifndef CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_SAVE_CRASH_TO_SETTINGS
 	DiagnosticLogProvider() = default;
-#endif
+
+	CHIP_ERROR SetIntentImplementation(chip::app::Clusters::DiagnosticLogs::IntentEnum intent,
+					   IntentData &intentData);
+	void DeleteIntentImplementation(chip::app::Clusters::DiagnosticLogs::IntentEnum intent, IntentData &intentData);
```

Comment:
You can make them private, or add a description if you want to keep them public.

Reply:
Done